ext/linux/osq_lock.h

```diff
@@ -78,22 +81,17 @@ void osq_parse_args(test_args unused, int argc, char** argv) {
  * spinning.
  */
 
-struct optimistic_spin_node * osq_nodepool_ptr;
-int random_sleep;
-
 static inline void osq_lock_init(uint64_t *lock, unsigned long cores)
 {
     /* calloc will set memory to zero automatically */
-    struct optimistic_spin_queue *osq_queue_ptr = calloc(1, sizeof(struct optimistic_spin_queue));
-    osq_nodepool_ptr = calloc(cores, sizeof(struct optimistic_spin_node));
-    if (osq_queue_ptr == NULL || osq_nodepool_ptr == NULL) exit(errno);
+    global_osq_nodepool_ptr = calloc(cores + 1, sizeof(struct optimistic_spin_node));
+    if (global_osq_nodepool_ptr == NULL) exit(errno);
 
     srand(time(0));
-    random_sleep = rand() % 100;
-    atomic_set(&osq_queue_ptr->tail, OSQ_UNLOCKED_VAL);
+    for (int i = 0; i < cores; i++)
+        (global_osq_nodepool_ptr + i)->random_sleep = rand() % 5 + 1;  /* 1 ~ 5 */
```

Comment:
Please describe the magic numbers used here, or better yet, define them as global consts.